src/main/java/com/zenith/util/math/MathHelper.java

```diff
@@ -44,4 +44,20 @@ public static float wrapDegrees(float degrees) {
     public static float clamp(final float value, final float min, final float max) {
         return Math.max(min, Math.min(max, value));
     }
+
+    public static double[] translateEntity(double x, double y, double z, float yaw, float pitch, double distance) {
+        double yawRadians = Math.toRadians(yaw);
+        double pitchRadians = Math.toRadians(pitch);
+
+        double xOffset = -Math.sin(yawRadians) * Math.cos(pitchRadians) * distance;
+        double yOffset = -Math.sin(pitchRadians) * distance;
+        double zOffset = Math.cos(yawRadians) * Math.cos(pitchRadians) * distance;
+
+        double newX = x + xOffset;
+        double newY = y + yOffset;
+        double newZ = z + zOffset;
+
+        return new double[]{newX,newY,newZ};
+    }
```

Comment:
I'd prefer inlining this to where its needed so we avoid heap allocations on the hotpath of packet processing. 

I don't think this func is being reused or shared anywhere else

Reply:
change was made